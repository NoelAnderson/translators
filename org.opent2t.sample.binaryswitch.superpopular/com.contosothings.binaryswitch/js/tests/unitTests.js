var test = require('ava');
var translatorPath = require('path').join(__dirname, '..');
var runBinarySwitchTests = require('opent2t-device-binaryswitch/binarySwitchTests');
var deviceData = require('./devicedata');
var OpenT2TLogger = require('opent2t').Logger;
var logger = new OpenT2TLogger("info");
var MockHub = require('opent2t-device-mockcontosothingshub/mockContosoThingsHub');
var mockHub = new MockHub(logger, deviceData);

var settings = {
    logger,
    test,
    createTranslator: mockHub.createTranslator(translatorPath, deviceData.base_state.Id),
    setTestData: mockHub.setTestData
};

// Run standard binary switch unit tests
runBinarySwitchTests(settings);