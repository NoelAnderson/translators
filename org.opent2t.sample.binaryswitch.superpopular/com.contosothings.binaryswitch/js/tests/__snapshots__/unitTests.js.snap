exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.binaryswitch.superpopular\",\"translator\":\"opent2t-translator-com-contosothings-binaryswitch\",\"controlId\":\"fb8e4fef-092c-486f-80d7-502a2715dfb7\"},\"availability\":\"online\",\"pi\":\"fb8e4fef-092c-486f-80d7-502a2715dfb7\",\"mnmn\":\"ContosoThings\",\"mnmo\":\"ctSwitch1\",\"n\":\"Backyard Lights\",\"rt\":[\"org.opent2t.sample.binaryswitch.superpopular\"],\"entities\":[{\"n\":\"Backyard Lights\",\"rt\":[\"oic.d.smartplug\"],\"di\":\"17183a67-ba07-190b-3241-44a283114f81\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/power\",\"rt\":[\"oic.r.switch.binary\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.binaryswitch.superpopular","translator":"opent2t-translator-com-contosothings-binaryswitch","controlId":"fb8e4fef-092c-486f-80d7-502a2715dfb7"},"availability":"online","pi":"fb8e4fef-092c-486f-80d7-502a2715dfb7","mnmn":"ContosoThings","mnmo":"ctSwitch1","n":"Backyard Lights","rt":["org.opent2t.sample.binaryswitch.superpopular"],"entities":[{"n":"Backyard Lights","rt":["oic.d.smartplug"],"di":"17183a67-ba07-190b-3241-44a283114f81","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/power","rt":["oic.r.switch.binary"],"if":["oic.if.a","oic.if.baseline"],"id":"power","value":false}]}]}"`;
